lib/poker_hands/hand_rankings/royal_flush.rb

```diff
@@ -1,5 +1,14 @@
 class RoyalFlush
   def check(hand)
-    true
+    royal_flush_ranks = ['A', 'K', 'Q', 'J', '10']
+    ranks_in_hand = []
+    
+    return false if hand.any? { |card| card.suit != hand.first.suit}
+
+    hand.size.times do |card|
+      ranks_in_hand << hand[card].rank
+    end
```

Comment:
Rubyism for you (assuming that `hand` is an array of `Card` objects)
```
hand.each do |card|
  ranks_in_hand << card.rank
end

# or

ranks_in_hand = hand.map(&:rank)
```

Reply:
Right on